src/ReplayingMessages/ReplayMessages.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EventSauce\EventSourcing\ReplayingMessages;
+
+use EventSauce\EventSourcing\MessageConsumer;
+use EventSauce\EventSourcing\MessageRepository;
+use EventSauce\EventSourcing\PaginationCursor;
+
+class ReplayMessages
+{
+    public function __construct(
+        private MessageRepository $repository,
+        private MessageConsumer $consumer,
+    ) {
+    }
+
+    public function replayBatch(PaginationCursor $cursor): ReplayResult
+    {
+        if ($cursor->isAtStart() && $this->consumer instanceof TriggerBeforeReplay) {
```

Comment:
One thing I'm not 100% sure of. Should the cursor tell `ReplayMessages` what the start is? Or should ReplayMessages implementation be responsible for knowing when to call beforeReplay? 

(Eg by having a `startReplay()` and a `continueReplay()` method?). 

Just a small thought, no big preference to one or the other way.